R/start_rains.R

```diff
@@ -1,71 +1,106 @@
+#' @param evaporation_variable \code{character(1)} Name of evaporation (mm) column, if using \code{"evaporation"}. Default \code{NULL}.
+#' @param fraction \code{numeric(1)} Multiplier (0–1) applied to evaporation values when \code{total_rainfall_comparison="evaporation"}. Default \code{0.5}.
+#' @param number_rain_days \code{logical(1)} If \code{TRUE}, define start by minimum rainy days in an interval. Default \code{FALSE}.
+#' @param min_rain_days \code{integer(1)} Minimum count of rainy days if \code{number_rain_days=TRUE}. Default \code{1}.
+#' @param rain_day_interval \code{integer(1)} Interval length (days) for \code{number_rain_days}. Default \code{2}.
+#' @param dry_spell \code{logical(1)} If \code{TRUE}, define start by a maximum dry‐spell length. Default \code{FALSE}.
+#' @param spell_interval \code{integer(1)} Interval length (days) for the dry‐spell test. Default \code{21}.
+#' @param spell_max_dry_days \code{integer(1)} Maximum consecutive dry days if \code{dry_spell=TRUE}. Default \code{9}.
+#' @param dry_period \code{logical(1)} If \code{TRUE}, define start by combining max rainfall and max dry days in a period. Default \code{FALSE}.
+#' @param period_interval \code{integer(1)} Interval length (days) for \code{dry_period}. Default \code{45}.
+#' @param max_rain \code{numeric(1)} Maximum total rainfall (mm) allowed in \code{period_interval} if \code{dry_period=TRUE}. Default \code{40}.
+#' @param period_max_dry_days \code{integer(1)} Maximum dry days if \code{dry_period=TRUE}. Default \code{30}.
+#' @param data_book A \code{DataBook} object to use. If \code{NULL}, a new one is created. Default \code{NULL}.
+#'
+#' @return Invisibly returns the used \code{data_book}.  The side effect is that a new
+#'   linked data frame (e.g.\ \code{<data>_by_<station>_<year>}) is created with columns
+#'   \code{start_rain}, \code{start_rain_date}, and/or \code{start_rain_status}.
 #'
-#' @examples #TODO#
-#' # check against R-Instat function
+#' @export
+#' 
+#' @examples
+#' library(databook)
+#' data_book <- DataBook$new()
+#' daily_data <- rpicsa::daily_niger %>%
+#'   dplyr::filter(year <= 1950) %>%
+#'   dplyr::filter(year > 1945) %>%
+#'   dplyr::mutate(year = as.numeric(year)) %>%
+#'   dplyr::filter(station_name == "Agades")
+#' data_book$import_data(list(daily_data = daily_data))
+#' 
+#' daily_data <- data_book$get_data_frame("daily_data")
+#' 
+#' suppressWarnings(start_rains(data = "daily_data",
+#'                              date_time = "date",
+#'                              station = "station_name",
+#'                              year = "year",
+#'                              rain = "rain",
+#'                              start_day = 121,
+#'                              end_day = 300,
+#'                              output = c("doy", "date", "status"),
+#'                              data_book = data_book))
+#' daily_data_by_station_name_year <- data_book$get_data_frame("daily_data_by_station_name_year.rds")
```

Comment:
@Emmanuel-Afrifa there is test data available to you already in the package. Try running this bit here from the examples part of the R file. It should work for you for the start of rains here to see some output. Let me know how you get on and if you understand it, as this just needs simple changes to then try out in other functions 
(If you look at end_seasons/end_rains I believe there should be examples there already)